woocommerce/woocommerce-ios
# Shows Deprecated Stats View after September 1st, 2020

closes https://github.com/woocommerce/woocommerce-ios/issues/2463

# Why
After September 1st of 2020, the dashboardV3 won't be supported anymore and [should be deleted](https://github.com/woocommerce/woocommerce-ios/issues/2522). This PR shows an empty state screen for that scenario indicating the user to update their WC version to gain access to the supported stats.

# How
- Modify `EmptyStateViewController.Config` to support a new style(`text`) of an actionable button.
- Adds a new `DeprecatedDashboardStatsViewController` to serve as the empty state screen.
- Modify `DashboardUIFactory` to serve the new empty state screen instead of `dashboardV3` after September 1st, 2020 

# Screenshots

Regular Empty State | Before 1st of September
----|----
<img width="413" alt="normal empty" src="https://user-images.githubusercontent.com/562080/87720613-d0b9da80-c77a-11ea-8e8c-a19e95667c54.png">|<img width="419" alt="stats-banner" src="https://user-images.githubusercontent.com/562080/87720593-c992cc80-c77a-11ea-9059-7c3e7cadf6be.png">


After 1st of September | After 1st of September Dark
----|----
<img width="415" alt="deprecated-view" src="https://user-images.githubusercontent.com/562080/87720606-ce578080-c77a-11ea-972d-a0a483b504af.png">|<img width="415" alt="deprecated dark" src="https://user-images.githubusercontent.com/562080/87720610-cf88ad80-c77a-11ea-82a9-91ec3d33752a.png">


# Testing steps
- Launch app with a site with an older version of WC or [disable WC-Admin using this plugin](https://wordpress.org/plugins/disable-dashboard-for-woocommerce/)
- See that the deprecated stats banner is shown

- Advance device clock after Sept 1st (and enable insecure connections temporarily)  **OR** Move the deprecated stats trigger date in `DashboardUIFactory` to a date before the current date

- See that the deprecated stats screen is shown

PS: If you choose to move the device clock, this is what needs to go in the `info.plist` in order to bypass SSL errors for having a different system date.
```xml
<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSAllowsLocalNetworking</key>
		<true/>
		<key>NSAllowsArbitraryLoads</key>
		<true/>
		<key>NSExceptionDomains</key>
		<dict>
			<key>wordpress.com</key>
			<dict>
				<key>NSIncludesSubdomains</key>
				<true/>
				<key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
				<true/>
				<key>NSTemporaryExceptionMinimumTLSVersion</key>
				<string>TLSv1.1</string>
			</dict>
		</dict>
	</dict>
```

<img width="412" alt="security" src="https://user-images.githubusercontent.com/562080/87722949-9fdba480-c77e-11ea-9961-5166d99d250c.png">



Update release notes:
